In [1]:
import os
import json
import pandas as pd
import csv

In [64]:
main_folder_path = './Compound_121475001_121500000/' #####여기 수정#########

data_list = []

for i in range(121475001, 121500000):    #####여기 수정########
    subfolder_name = f"{i:09d}"
    subfolder_path = os.path.join(main_folder_path, subfolder_name)

    json_file_name = f"{subfolder_name}.B3LYP@PM6.S0.json"
    json_file_path = os.path.join(subfolder_path, json_file_name)

    if os.path.exists(json_file_path):
        with open(json_file_path, 'r') as json_file:
            json_data = json.load(json_file)

            molecular_formula = json_data["pubchem"]["molecular formula"]
            molecular_weight = json_data["pubchem"]["molecular weight"]
            atom_count = json_data["pubchem"]["B3LYP@PM6"]["atoms"]["elements"]["atom count"]
            heavy_atom_count = json_data["pubchem"]["B3LYP@PM6"]["atoms"]["elements"]["heavy atom count"]
            total_dipole_moment = json_data["pubchem"]["B3LYP@PM6"]["properties"]["total dipole moment"]
            homo_alpha = json_data["pubchem"]["B3LYP@PM6"]["properties"]["energy"]["alpha"]["homo"]
            lumo_alpha = json_data["pubchem"]["B3LYP@PM6"]["properties"]["energy"]["alpha"]["lumo"]
            gap_alpha = json_data["pubchem"]["B3LYP@PM6"]["properties"]["energy"]["alpha"]["gap"]
            isomeric_smiles = json_data["pubchem"]["Isomeric SMILES"]

            # Append the extracted information as a dictionary to the data_list
            data_list.append({
                "i": i,
                "molecular_formula": molecular_formula,
                "molecular_weight" : molecular_weight,
                "atom_count": atom_count,
                "heavy_atom_count": heavy_atom_count,
                "total_dipole_moment": total_dipole_moment,
                "homo": homo_alpha,
                "lumo": lumo_alpha,
                "gap": gap_alpha,
                "Isomeric_SMILES": isomeric_smiles
            })

# Create a pandas DataFrame from the data_list
df = pd.DataFrame(data_list)

# Set the "i" column as the index of the DataFrame
df.set_index('i', inplace=True)


In [65]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3220 entries, 121475066 to 121494105
Data columns (total 9 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   molecular_formula    3220 non-null   object 
 1   molecular_weight     3220 non-null   float64
 2   atom_count           3220 non-null   int64  
 3   heavy_atom_count     3220 non-null   int64  
 4   total_dipole_moment  3220 non-null   float64
 5   homo                 3220 non-null   float64
 6   lumo                 3220 non-null   float64
 7   gap                  3220 non-null   float64
 8   Isomeric_SMILES      3220 non-null   object 
dtypes: float64(5), int64(2), object(2)
memory usage: 251.6+ KB


In [66]:
output_csv_file = './Compound_121475001_121500000.csv' #####여기 수정#########
df.to_csv(output_csv_file)

__각 폴더별 데이터개수__ 

Compound_000000001_000025000 : 12683 <br>
Compound_000025001_000050000 : 9449 <br>
Compound_000050001_000075000 : 9489 <br>
Compound_020000001_020025000 : 251<br>
Compound_020050001_020075000 : 4902<br>
Compound_022475001_022500000 : 4672 <br>
Compound_030000001_030025000 : 92 <br>
Compound_040000001_040025000 : 1244 <br>
Compound_040050001_040075000 : 1606<br>
Compound_050725001_050750000 : 955<br>
Compound_060000001_060025000 : 4523 <br>
Compound_060050001_060075000 : 4216 <br>
Compound_070000001_070025000 : Error<br>
Compound_080000001_080025000 : 21033 <br>
Compound_080050001_080075000 : 19836 <br>
Compound_090000001_090025000 : 4854 <br>
Compound_100000001_100025000 : 3851 <br>
Compound_100050001_100075000 : 2867 <br>
Compound_110000001_110025000 : 9808 <br>
Compound_120000001_120025000 : 2174 <br>
Compound_120050001_120075000 : 2272 <br>
Compound_121475001_121500000 : 3220 <br>
